aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1360,7 +1360,8 @@ static inline int64_t get_mkldnn_matmul_min_dim() {
       // Minimum dimension requirement for MKLDNN; derived based on experiments.
       // By default, it's only enabled on Neoverse V1.
 #if !defined(__s390x__)  && !defined(__powerpc__)
-      if (cpuinfo_initialize() && cpuinfo_get_uarchs_count() == 1 && cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v1) {
+      if (cpuinfo_initialize() && cpuinfo_get_uarchs_count() == 1 &&
+          (cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v1 || cpuinfo_get_uarch(0)->uarch == cpuinfo_uarch_neoverse_v2)) {
```

Comment:
Can you please elaborate why this heuristic should be applied to neonvese_v2 but not to neonverse_n2?
Also, considering the code is almost identical between `get_mkldnn_matmul_min_dim` and `get_mkldnn_matmul_min_size` do you mind refactoring this into `is_arm_neonverse()` helper routine